chrome/browser/android/adblock/regex_resource_type_detector.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * This file is part of Adblock Plus <https://adblockplus.org/>,
+ * Copyright (C) 2006-present eyeo GmbH
+ *
+ * Adblock Plus is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * Adblock Plus is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Adblock Plus.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef CHROME_BROWSER_ANDROID_REGEX_RESOURCE_TYPE_DETECTOR_H_
+#define CHROME_BROWSER_ANDROID_REGEX_RESOURCE_TYPE_DETECTOR_H_
+
+#include <string>
+#include "content/public/common/resource_type.h"
+
+class RegexResourceTypeDetector {
+  public:
+    // returns `content::ResourceType::RESOURCE_TYPE_LAST_TYPE` for unknown resource types
+    content::ResourceType detect(std::string url);
```

Comment:
i personally still prefer separate class but i will do what most of us agree on.
so you vote for method, not class. right?

Reply:
I'm with sergei here. I don't think there is a need for a class. A simple function would be enough. But if you really wanted a class, this method should be a class method (ie `static`) because at no point it needs `this`; this would also make the change in `chrome_network_delegate.h` unnecessary.

Reply:
No problem, i will introduce just `static` method